File: python/알고리즘/baek/브루트포스/2531_회전초밥.c
#include <stdio.h>
#define max(a,b)   (((a) > (b)) ? (a) : (b))

int N, D, K, C;
int Sushis[30000+10];

void InputData(void){
    scanf("%d %d %d %d", &N, &D, &K, &C);
    for (int i=0; i<N ; i++){
        scanf("%d", &Sushis[i]);
        // printf("%d ", Sushis[i]);
    }
}

int main(void) {

    InputData();
    
    int result = 0;

    for (int i = 0; i < N; i++) {
        int check[3010] = {0,};
        int cnt = 0;
        for (int j = i; j < i+K; j++) {
            int p = j;
            if (j >= N) {
                p -= N;
            }
            int sushi = Sushis[p];

            if (check[sushi]) {
                continue;
            }

            check[sushi] = 1;
            cnt++;
        }

        if (check[C] == 0) {
            cnt++;
        }
        result = max(result, cnt);
    }

    printf("%d", result);
    return 0;
}

File: python/알고리즘/baek/브루트포스/2531_회전초밥.py
N,d,k,c = map(int, input().split())

sushis = [int(input()) for _ in range(N)]

# 초밥 하나씩 시작해서 먹어보기
result = 0
for i in range(N):
    check = [0] * 3001
    cnt = 0
    for j in range(i, i+k, 1):
        if j >= N:
            j -= N
        sushi = sushis[j]

        if check[sushi]:
            continue
        check[sushi] = 1
        cnt += 1
    if check[c] == 0:
        cnt += 1
    
    result = max(result, cnt)
print(result)
